### ISSUE TITLE
Radar Cross Section (RCS) Generates with strange angles

### ISSUE THREAD

**Issue by Molthus** [PapaJoesSoup/BDArmory#711]
##### ISSUE TYPE

 - Support 	    ( Need help getting something to work)
##### KSP and BDA Version

```KSP 1.7.2 (x64)```

```BDA 1.3.1.0```

##### OS / ENVIRONMENT

Windows 10

##### SUMMARY
May be related to https://github.com/PapaJoesSoup/BDArmory/issues/575, not sure.

Radar Cross Section appears to generate at strangely skewed angles, best demonstrated with images. I may be misinterpreting how the system is intended to function, but it seems to have strange behavior in regards to the origin points of the RCS detection.

Screenshots:
As expected:
![screenshot0](https://user-images.githubusercontent.com/52179243/60053517-ac85ea00-969d-11e9-9f89-4014df91f75c.png)

But why does this render the way it does (especially the lateral)?
![screenshot1](https://user-images.githubusercontent.com/52179243/60053547-bc053300-969d-11e9-8bb3-afd80b6db718.png)

This is where this moves from appearing to be a part of the mechanics to looking like a bug. Where does the ventral get the idea to render two full panels?
![screenshot3](https://user-images.githubusercontent.com/52179243/60053585-d9d29800-969d-11e9-91a1-d461cd2388f1.png)
If the system allows it to accept rays cast by the front, the freshly added panel here should block that, giving the ventral view no reflection.

Can see rotation strangeness here as well; not sure if this is just rotating the image to fit the windows better or if the cameras/casting points actually rotated (it feels like the latter)
![screenshot6](https://user-images.githubusercontent.com/52179243/60053685-1dc59d00-969e-11e9-824c-cb2a80f1558b.png)

![screenshot9](https://user-images.githubusercontent.com/52179243/60053739-48175a80-969e-11e9-99f7-55c230a685c5.png)

I have more screenshots but I think this makes the point. This feels like a bug to me, but I've marked this as support because I don't know enough about how the system is supposed to work to know it is a bug. As far back as I can remember, this system has been doing this, making the creation of stealth vehicles on angles alone feel rather impossible.

Any help would be greatly appreciated!

##### STEPS TO REPRODUCE

1.) Install KSP 1.7.2, BDA 1.3.1.0, and Physics Range Extender 1.12.
2.) Go to VAB
3.) Create shapes in above screenshots
4.) Open and view Radar Cross Section

##### Log Files
[KSP.log](https://github.com/PapaJoesSoup/BDArmory/files/3322681/KSP.log)
(No Output.txt found)

**Comment by Molthus** [comment]
I just realized Output.txt was probably meant to be output_log.txt which I found and have included here; apologies for not including it before.
[output_log.txt](https://github.com/PapaJoesSoup/BDArmory/files/3322784/output_log.txt)

**Comment by Gedas-S** [comment]
From the top of my head currently the calculation of cross-section is based on the root part, so rotating the root part will do nothing for angles. So, for example, the ventral cross-section is looking from the bottom of the craft (based on root part rotation), not from the floor of the SPH.

Additionally, iirc cross-section is calculated in six directions, each of the three directions displayed also calculates another cross-section rotated by 45 degrees, and takes the higher of the two. Which is what I believe you're seeing in the lateral image of images 2 and 3.

**Comment by Molthus** [comment]
Sorry for the delay and thanks for the response!

Further testing indicates that you are correct. Unfortunately, after a cumulative several hours, I just cannot find ways of making viable stealth designs with the current system. Every attempt to make angles that divert radar winds up making it worse (because of the alternate angles) and the best ways seem to involve really gimpy techniques like placing odd fins and in general cheesing the system rather than being able to make a viable design. It seems right now that the only way to make a stealth plane is just to make it absolutely tiny and couple it with jammers, which is disappointing to me, however, if you are aware of any planes that are stealth and don't just rely on being tiny or on jammers, or if you have any other recommendations as to how I can go about and actually make a stealth plane with the current system besides making it tiny or jam, please let me know. I've looked extensively and haven't really found anything.

Thanks again!

**Comment by Gedas-S** [comment]
You are correct. I don't think the current system will allow you to create an angle-based stealth vehicle.

It's one of the things that I've been thinking might be double though, so.... maybe it'll happen in the future. ;)

**Comment by Gh0stReaper123** [comment]
> Additionally, iirc cross-section is calculated in six directions, each of the three directions displayed also calculates another cross-section rotated by 45 degrees, and takes the higher of the two.

Looking through the code to try and find #575, I can confirm. It's on line 280 of RadarUtils.cs.
